pkg/kubectl/cmd/util/factory.go

```diff
@@ -441,7 +441,7 @@ func (f *Factory) PrintObject(cmd *cobra.Command, obj runtime.Object, out io.Wri
 
 // PrinterForMapping returns a printer suitable for displaying the provided resource type.
 // Requires that printer flags have been added to cmd (see AddPrinterFlags).
-func (f *Factory) PrinterForMapping(cmd *cobra.Command, mapping *meta.RESTMapping, withNamespace bool) (kubectl.ResourcePrinter, error) {
+func (f *Factory) PrinterForMapping(cmd *cobra.Command, mapping *meta.RESTMapping, withNamespace bool, withHeader bool) (kubectl.ResourcePrinter, error) {
```

Comment:
While we're refactoring this area, let's clean up this call.  I see a context object of some sort containing:
```
type PrintingOptions struct{
    withNamespace bool
    withHeaders bool
    wideFormat bool
    labelColumns []string
    sortBy string
}
```

You could have a helper for building that from `cobra.Command`, but I don't see a reason for this object to be tied a `cobra.Command`.